src/transactions/feed/EarnFeedItem.tsx

```diff
@@ -0,0 +1,160 @@
+import { getDynamicConfigParams } from 'src/statsig'
+import { DynamicConfigs } from 'src/statsig/constants'
+import { StatsigDynamicConfigs } from 'src/statsig/types'
+import { Colors } from 'src/styles/colors'
+import { typeScale } from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+import variables from 'src/styles/variables'
+import TransactionFeedItemImage from 'src/transactions/feed/TransactionFeedItemImage'
+import {
+  EarnClaimReward,
+  EarnDeposit,
+  EarnWithdraw,
+  TransactionStatus,
+} from 'src/transactions/types'
+
+interface AmountDisplayProps {
+  transaction: EarnWithdraw | EarnDeposit | EarnClaimReward
+  isLocal: boolean
+}
+
+function AmountDisplay({ transaction, isLocal }: AmountDisplayProps) {
+  let amountValue
+  let tokenId
+
+  switch (transaction.__typename) {
+    case 'EarnDeposit':
+      amountValue = new BigNumber(-transaction.inAmount.value)
+      tokenId = transaction.outAmount.tokenId
+      break
+    case 'EarnWithdraw':
+      amountValue = new BigNumber(transaction.outAmount.value)
+      tokenId = transaction.inAmount.tokenId
+      break
+    case 'EarnClaimReward':
+      amountValue = new BigNumber(transaction.amount.value)
+      tokenId = transaction.amount.tokenId
+      break
+  }
+
+  const textStyle = isLocal
+    ? styles.amountSubtitle
+    : [styles.amountTitle, transaction.__typename !== 'EarnDeposit' && { color: Colors.primary }]
+
+  return (
+    <TokenDisplay
+      amount={amountValue}
+      tokenId={tokenId}
+      showLocalAmount={isLocal}
+      showSymbol={true}
+      showExplicitPositiveSign={!isLocal}
+      hideSign={!!isLocal}
+      style={textStyle}
+      testID={`EarnFeedItem/${transaction.__typename}-amount-${isLocal ? 'local' : 'crypto'}`}
+    />
+  )
+}
+
+interface AmountProps {
+  transaction: EarnWithdraw | EarnDeposit | EarnClaimReward
+}
+
+function Amount({ transaction }: AmountProps) {
+  return (
+    <View style={styles.amountContainer}>
+      <AmountDisplay transaction={transaction} isLocal={false} />
+      <AmountDisplay transaction={transaction} isLocal={true} />
+    </View>
+  )
+}
+
+interface Props {
+  transaction: EarnWithdraw | EarnDeposit | EarnClaimReward
+}
+
+export default function EarnFeedItem({ transaction }: Props) {
+  const { t } = useTranslation()
+  const { providerName } = getDynamicConfigParams(
+    DynamicConfigs[StatsigDynamicConfigs.EARN_STABLECOIN_CONFIG]
+  )
+  const isDeposit = transaction.__typename === 'EarnDeposit'
+  const isWithdraw = transaction.__typename === 'EarnWithdraw'
+  const isClaim = transaction.__typename === 'EarnClaimReward'
+
+  return (
+    <Touchable
+      testID={`EarnFeedItem/${transaction.transactionHash}`}
+      onPress={() => {
+        ValoraAnalytics.track(EarnEvents.earn_feed_item_select, { origin: transaction.__typename })
+        navigate(Screens.TransactionDetailsScreen, {
+          transaction: { ...transaction, status: TransactionStatus.Complete },
```

Comment:
is there a reason to hardcode complete here? I have a few that have failed already 😄

Reply:
It seems that we're not getting a status back from blockchain-api currently 🤔

Reply:
Fixed in ceeaefcf3d715dcc02c444e1979e5ae709c8b127!